src/typing/typeloadFields.ml

```diff
@@ -763,7 +763,24 @@ let bind_var (ctx,cctx,fctx) cf e =
 					error ("Redefinition of variable " ^ cf.cf_name ^ " in subclass is not allowed. Previously declared at " ^ (s_type_path csup.cl_path) ) p
 	end;
 	let t = cf.cf_type in
-
+	let e =
+		let rec default_value t =
+			match t with
+			| TMono { tm_type = Some t } -> default_value t
+			| TLazy { contents = LAvailable t } -> default_value t
+			| TType({ t_type = t },_) -> default_value t
+			| TAbstract({ a_path = [],"Null" },_) -> None
+			| TAbstract({ a_path = [],("Int"|"Float") },[]) -> Some ((EConst (Int "0")),cf.cf_pos)
```

Comment:
It's better to have separate cases for these so we don't run into weird Int vs. Float situations. I think at least neko does something quite strange there.